caas/containers.go

```diff
@@ -50,13 +50,39 @@ type ContainerSpec struct {
 
 	// ProviderContainer defines config which is specific to a substrate, eg k8s
 	ProviderContainer `yaml:"-"`
+
+	Constraints []Constraint `yaml:"constraint,omitempty"`
+}
+
+type DeviceType string
+
+// Constraint defines resource limit/request for pod.
+type Constraint struct {
+	Type       DeviceType        `bson:"type"`
+	Count      int64             `bson:"count"`
+	Attributes map[string]string `bson:"attributes"`
+}
+
+// AffinitySpec defines affinity configuration.
+type AffinitySpec struct {
+	// TODO(ycliuhw): enrich spec here later for supporting node affinity.
+	NodeLabel NodeLabel `yaml:"nodeLabel,omitempty"`
+}
+
+// NodeLabel defines node label to use for nodeSelector.
+type NodeLabel struct {
+	// TODO(ycliuhw): enrich spec here later for specifying node to deploy to in AffinitySpec.
+
+	// Name maps to k8s accelerator name
+	Name string `yaml:"name"`
 }
 
 // PodSpec defines the data values used to configure
 // a pod on the CAAS substrate.
 type PodSpec struct {
 	Containers          []ContainerSpec `yaml:"-"`
 	OmitServiceFrontend bool            `yaml:"omitServiceFrontend"`
+	AffinitySpec        *AffinitySpec   `yaml:"affinitySpec,omitempty"`
```

Comment:
This doesn't go here.

Take a look at what we do for storage - we create the k8s pod spec with the info we get from the charm, then we separately call a configureStorage() func to add in the storage bits.